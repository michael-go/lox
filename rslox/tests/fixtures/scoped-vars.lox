var g1 = "foo";

{
    var l1 = g1 + "bar";
    print l1;
    
    {
        var l1 = "shadowed";
        print l1;
    }
}

print g1;
// TODO: is it ok these print nil rather than error?
print l1;
print no_such_var;