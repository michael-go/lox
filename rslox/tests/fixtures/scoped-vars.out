# exit code: 70
# stdout:
foobar
shadowed
foo

# stderr:
Undefined variable 'l1'.
[line 15] in <script>

